wren-ui/src/apollo/server/managers/dataSourceSchemaDetector.ts

```diff
@@ -122,21 +150,17 @@ export default class DataSourceSchemaDetector
 
     const modelRelationships =
       await this.ctx.relationRepository.findRelationInfoBy({
-        columnIds: modelColumns.map((column) => column.id),
+        modelIds,
       });
 
     // Handle resolve deleted tables
     if (schemaChangeType === SchemaChangeType.DELETED_TABLES) {
       await Promise.all(
         affectedModels.map(async (model) => {
-          const affectedColumns = changes.find(
-            (table) => table.name === model.sourceTableName,
-          ).columns;
-
-          // Get affected calculated fields and affected relationships
+          // Get affected columns and calculated fields and affected relationships
           const affectedMaterials = await this.getAffectedResources(
```

Comment:
```javascript
 // Handle resolve deleted tables
    if (schemaChangeType === SchemaChangeType.DELETED_TABLES) {
      await Promise.all(
        affectedModels.map(async (model) => {
          // Get affected columns and calculated fields and affected relationships
          const affectedMaterials = await this.getAffectedResources(
            model,
            changes,
            {
              models,
              modelColumns,
              allCalculatedFields,
              modelRelationships,
            },
          );
     ....
}
```
I'm wondering if we want to get all the affected resources of all affected models, and the affected models come from changes (L141).
Maybe we do not need to get the affected materials by models, get all the affected Resources at once.
```suggestion
const affectedMaterials = await this.getAffectedResources(
            changes,
            {
              models,
              modelColumns,
              modelRelationships,
            },
          );
```